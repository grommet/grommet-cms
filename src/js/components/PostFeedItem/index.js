/* @flow */
import React from 'react';
import Box from 'grommet/components/Box';
import Anchor from 'grommet/components/Anchor';
import Heading from 'grommet/components/Heading';
import Section from 'grommet/components/Section';
import Label from 'grommet/components/Label';
import { formatPrettyDate } from 'grommet-cms/utils';

export default function PostFeedItem(props: {
  post: {
    title: string,
    image: Object,
    slug: string,
    createdAt: string
  },
  colorIndex: string,
  postPath: string,
}) {
  const { post, postPath, colorIndex } = props;
  return (
    <Anchor className="post-feed-item--anchor" path={`${postPath}${post.slug}`}>
      <Section
        full
        colorIndex={colorIndex}
        texture={post.image ? post.image.path : ''}
        pad="large"
      >
        <Box
          align="center"
          justify="start"
          pad={{ vertical: 'large' }}
          full
        >
          <Heading strong align="center">
            {post.title}
          </Heading>
          <Label className="post-feed-item--subtitle">
            {post.subtitle ? post.subtitle : ''}
          </Label>
          <Heading align="center" tag="h5">
            {`Posted on ${formatPrettyDate(post.createdAt)}`}
          </Heading>
        </Box>
      </Section>
    </Anchor>
  );
}
