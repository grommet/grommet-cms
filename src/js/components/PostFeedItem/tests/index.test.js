import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import PostFeedItem from '../index';
import props from './__mocks__/postFeedItemProps';

describe('<PostFeedItem />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <PostFeedItem {...props} />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
