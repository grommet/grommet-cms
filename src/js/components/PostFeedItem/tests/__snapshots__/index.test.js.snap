exports[`<PostFeedItem /> should render with default props 1`] = `
<Anchor
  animateIcon={true}
  className="post-feed-item--anchor"
  method="push"
  path="/post/watch-our-discover-video"
  tag="a">
  <Section
    colorIndex="grey-1-a"
    full={true}
    pad="large"
    texture="/uploads/media/2017/1/labs-blogfeed-1484268872525.gif">
    <Box
      align="center"
      announce={false}
      direction="column"
      focusable={true}
      full={true}
      justify="start"
      pad={
        Object {
          "vertical": "large",
        }
      }
      responsive={true}
      tag="div">
      <Heading
        align="center"
        strong={true}
        tag="h1">
        Watch our Discover Video
      </Heading>
      <Label
        className="post-feed-item--subtitle"
        size="medium" />
      <Heading
        align="center"
        tag="h5">
        Posted on January 16, 2017
      </Heading>
    </Box>
  </Section>
</Anchor>
`;
